pyteal/ast/router.py

```diff
@@ -508,19 +489,26 @@ def __init__(
         name: str,
         bare_calls: BareCallActions | None = None,
         descr: str | None = None,
+        *,
+        clear_state: Optional[ActionType] = None,
     ) -> None:
         """
         Args:
             name: the name of the smart contract, used in the JSON object.
             bare_calls: the bare app call registered for each on_completion.
             descr: a description of the smart contract, used in the JSON object.
+            clear_state: an expression describing the behavior of clear state program.
         """
 
         self.name: str = name
         self.descr = descr
 
         self.approval_ast = ASTBuilder()
-        self.clear_state_ast = ASTBuilder()
+        self.clear_state: Expr = (
+            Reject()
+            if clear_state is None
+            else ASTBuilder.wrap_handler(False, clear_state)
```

Comment:
nit: right now `wrap_handler` has error messages that talk about bare app calls. That will no longer be accurate for the clear state program, since it's not a normal bare app call.

Reply:
made a push on [2575241](https://github.com/algorand/pyteal/pull/636/commits/2575241e66155778c6a41865e4793e25ce27dbea), so error message varies by provided args